src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -205,8 +204,16 @@ def _create_task_list(
     program_threads: number of threads to use per task
     For instance, if "threads" is set to 128 and "program_threads" to 32, then each task will run 4 batches in parallel.
 
-    We have chosen to use "taskfarmer" for parallelization, which means that "threads" and "program_threads" should
-    have the same value. This ensures that parallelization only happens between tasks, and not within them.
+    For tools capable of processing a batch of genomes, such as GTDB-TK and checkm2,
+    we have chosen to utilize the THREADS variable within Taskfarmer batch script (submit_taskfarmer.sl) for parallelization,
+    which means that "threads" and "program_threads" should have the same value.
+    This ensures that parallelization only happens between tasks, and not within them.
+
+    For tools that can only handle genomes individually, such as microtrait, mash and eggNOG,
+    we still use the THREADS variable for task parallelization. However, within each task, we further parallelize tool
+    execution using the 'threads' variable. For instance, with 'threads' set to 16 and 'tasks_per_node' set to 4,
+    each node will concurrently execute 4 tasks, with each task executing 16 parallel tool operations.
```

Comment:
I thought you said you manually set NODE_THREADS to 1 for these tools

Reply:
just for eggNOG. Previously, we I run tools with collections data, I never need to manually update anything.

Reply:
it won't. I still need to manually update default `threads` and `program_threads` arg in `_create_task_list` to 16. I think those should be supplied by CLI.

Reply:
Ok, so this PR is just a small cleanup basically?

Reply:
yea. I haven't completely thought through how we want to treat `threads` and `NODE_THREADS_DEFAULT`

Reply:
yeah, let's talk though that at some point. I def have opinions on how this should work

Reply:
:+1: